shop-inc/identity-management
# #164457805 Add method to get the protocol buffers

#### What does this PR do?
Add a protocol buffers broker
#### Description of Task to be completed?
 - Add PROTOBUF_URL, PROTOBUF_USER and PROTOBUF_BRANCH environment
   variables
 - Add code to retrieve protocol buffers using https
 - add `.proto` files to `.gitignore`

#### How should this be manually tested?
1. Clone the origin repository [here](https://github.com/davenjeru/identity-management/).
2. Checkout to the branch `ft-identity-proto-broker-164457805`.
3 Add the necessary environment variables
4. Make sure `NODE_ENV` is set to `staging` or `production`.
5. Run `yarn install` to install dependencies.
6. Run `yarn test` for tests.
7. Run the gRPC server using `yarn start:dev`
8. The server should start with no errors
9. Close the server and inspect the generated `dist` folder. You sould see a `.proto` file called `identity.proto`

#### Any background context you want to provide?
The `.proto` files will only be fetched via https on staging and production environments. During development, it is encouraged that the `proto` files retrieved are copied to the `server/` directory.

#### What are the relevant pivotal tracker stories?
* [#164457805](https://www.pivotaltracker.com/n/projects/2316601/stories/164457805)
